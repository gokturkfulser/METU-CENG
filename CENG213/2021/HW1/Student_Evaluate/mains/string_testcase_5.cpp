#include <iostream>

#include "String.h"
//#include "../String.h"

/*
 * Case 5 : Default constructor; various inserts; various erasures; print.
 */
int main() {
    std::cout << "-> Creating a string with default constructor." << std::endl;

    String str;

    std::cout << "-> Appending and inserting some characters to the string." << std::endl;

    str.appendCharacter('e');
    str.appendCharacter('h');
    str.insertCharacter('m', 0);
    str.appendCharacter('b');
    str.insertCharacter('r', 2);
    str.appendCharacter('a');
    str.insertCharacter('a', 4);

    std::cout << "-> Printing the string." << std::endl;

    str.print(true);

    std::cout << "-> Erasing some characters and finally erasing all characters of the string. Printing the string after each erasing." << std::endl;

    std::cout << "-> eraseCharacters ..." << std::endl;

    str.eraseCharacters(0, 1);
    str.print(true);

    std::cout << "-> eraseCharacters ..." << std::endl;

    str.eraseCharacters(3, 1);
    str.print(true);

    std::cout << "-> eraseCharacters ..." << std::endl;

    str.eraseCharacters(1, 2);
    str.print(true);

    std::cout << "-> eraseAllCharacters ..." << std::endl;

    str.eraseAllCharacters();
    str.print(true);

    return 0;
}
