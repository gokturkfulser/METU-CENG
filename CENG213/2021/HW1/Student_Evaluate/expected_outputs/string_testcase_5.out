-> Creating a string with default constructor.
-> Appending and inserting some characters to the string.
-> Printing the string.
['m','e','r','h','a','b','a']
-> Erasing some characters and finally erasing all characters of the string. Printing the string after each erasing.
-> eraseCharacters ...
['e','r','h','a','b','a']
-> eraseCharacters ...
['e','r','h','b','a']
-> eraseCharacters ...
['e','b','a']
-> eraseAllCharacters ...
[]
